### ISSUE TITLE
Solved.

### ISSUE THREAD

**Issue by malekamiri** [codykerns/PurchasesHelper#9]
Nevermind.

**Comment by malekamiri** [comment]
Never mind
Seems like I'm just supposed to put 1.0 not 50. Not sure why the documentation said 50.

**Comment by codykerns** [comment]
Sorry for the confusion here. The documentation uses sample build numbers. Essentially, you need to use build numbers and not version numbers of the app versions you want to support. CFBundleVersion (build version) rather than CFBundleShortVersionString (release version).

https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleversion

**Comment by malekamiri** [comment]
Oh okay. Do you know where I could find a list of all my previous build numbers?

**Comment by malekamiri** [comment]
Version history on App Store Connect shows me all my previous versions but they are in this format: 1.0, 2.0.0, 2.1.0, 2.2.0 etc.

**Comment by malekamiri** [comment]
Build numbers repeat right? Like build number 1 will occur for version 1.0 and for version 2.0
Thanks for your help

I just printed CFBundleVersion of my app currently and it says 2. But I've had lots of builds so I am confused what number I am supposed to use.

**Comment by codykerns** [comment]
The "Activity" view in App Store Connect should show previous uploaded versions plus the build numbers- there's a cut off date for old versions, but I'm not sure what the cutoff is. It doesn't indicate which build was live on the App Store (if there were multiple builds for the same version, like TestFlight builds etc) but that should help!

**Comment by malekamiri** [comment]
So if I want users that downloaded 1.0 at build 2 to get premium features, I need to do this?
```
CompatibilityAccessManager.shared.register(entitlement:
    .init(entitlement: "premium_access", versions: ["2"])
)
```


But then if version 2.2 has build number 2 as well, wouldn't they also get access?


I guess I could start releasing odd build numbers (higher than usual) so they don't get supported?


Thanks

**Comment by codykerns** [comment]
Yep! You'll want to make sure you increment build numbers for each version or use some kind of separate versioning system to keep track of the builds.

**Comment by malekamiri** [comment]
Perfect thank you for your help!!

**Comment by malekamiri** [comment]
Ok last question. I just tested in sandbox with this code:
`CompatibilityAccessManager.shared.register(entitlement:
                                                    .init(entitlement: "Premium", versions: ["1.0"])
        )`

And it worked, user gets access to premium feature.

I tried with this:
`CompatibilityAccessManager.shared.register(entitlement:
                                                    .init(entitlement: "Premium", versions: ["50"])
        )`
And it didn't work, user does not get access to premium features.

I also tries with this:
`CompatibilityAccessManager.shared.register(entitlement:
                                                    .init(entitlement: "Premium", versions: ["1"])
        )`
And it didn't work, user does not get access to premium features.

Seems like putting "1.0" is what works in a sandbox environment?

Thanks again for your help! Very appreciated
